src/app/store/modules/river-index.js

```diff
@@ -0,0 +1,180 @@
+      Object.assign(state, { loading: false, userLocation: payload })
+    },
+    ['MAP_STYLE'](state, payload) {
+      Object.assign(state, { loading: false, mapStyle: payload })
+    },
+    ['MAP_COLOR_BY'](state, payload) {
+      Object.assign(state, { loading: false, mapColorBy: payload })
+    },
+    ['MAP_BOUNDS'](state, payload) {
+      Object.assign(state, { loading: false, mapBounds: payload })
+    },
+    ['MOUSEOVERED_FEATURE'](state, payload) {
+      Object.assign(state, { loading: false, mouseoveredFeature: payload })
+    },
+    ['MAP_SEARCH_TERM'](state, payload) {
+      Object.assign(state, { mapSearchTerm: payload })
+    },
+    ['MAP_SEARCH_LOADING'](state, payload) {
+      Object.assign(state, { mapSearchLoading: payload })
+    },
+    ['MAP_SEARCH_SUCCESS'](state, payload) {
+      Object.assign(state, { mapSearchResults: payload, mapSearchLoading: false })
+    },
+    ['MAP_SEARCH_ERROR'](state, payload) {
+      Object.assign(state, {
+        mapSearchLoading: false,
+        mapSearchResults: [],
+        error: payload || true
+      })
+    },
+  },
+  actions: {
+    ...actions,
+    fetchUserLocation(context, data) {
+      context.commit('USER_LOCATION', data)
+    },
+    loadReaches(context, data) {
+      context.commit('DATA_SUCCESS', data)
+    },
+    setMapStyle(context, data) {
+      context.commit('MAP_STYLE', data)
+    },
+    setMapColorBy(context, data) {
+      context.commit('MAP_COLOR_BY', data)
+    },
+    setMapBounds(context, data) {
+      context.commit('MAP_BOUNDS', data)
+    },
+    mouseOverFeature(context, data) {
+      context.commit('MOUSEOVERED_FEATURE', data)
+    },
+    async fetchStates(context) {
+      context.commit(type.DATA_LOADING)
+
+      // const result = await fetchStates(data).catch(e => {
+      //   context.commit(DATA_ERROR, e)
+      // })
+
+      const result = await httpClient.post('/graphql', {
+        query: `
+            query {
+                  states(first:100,aw_only:true){data{gmi,name,shortkey,type,aw_region,num_rivers,num_gauges}}
+            }
+        `
+      },
+        {
+          baseUrl: 'https://beta.americanwhitewater.org/'
+        })
+
+      if (result) {
+        context.commit('STATE_LIST', result.data.data.states.data
+          .filter(x => x.gmi.match(/USA-.*/))
+          .sort((a, b) => {
+            if (a.shortkey.toUpperCase() > b.shortkey.toUpperCase()) { return 1 }
+            if (a.shortkey.toUpperCase() < b.shortkey.toUpperCase()) { return -1 }
+            return 0
+          })
+        )
+      }
+      return result
+    },
+    async fetchMapSearchData(context, data) {
+      context.commit('MAP_SEARCH_TERM', data)
+
+      let result
+      // if there's no search term, just set results to empty array
+      if (!data || data.length === 0) {
+        result = []
+      } else {
+        context.commit('MAP_SEARCH_LOADING', true)
+        // result = await fetchRiverSearchData(data).catch(e => {
```

Comment:
how come you're embedding all the queries in here instead of importing from services like before?

Reply:
I'm struggling with where to put these. Importing them from `services/` that live in the rapid-detail directory, into the module file, kind of feels like spaghetti to me. 

Then I was thinking we could make a single controllers folder as a child of src, `@/controllers`, where we list all the API requests in one place then by view? `@/controllers/river-detail`...

Another idea was to pass the request to the dispatch to keep the files together declared in their directory and help with stubbing/mocking in tests.

```js
const apiRequest = () => {
  // http stuff
}

this.$store.dispatch('MyModule/getProperty', apiRequest)

async function getProperty(context, method) {

  try {
    context.commit('DATA_REQUEST')
    const result = await method()

    context.commit('DATA_SUCCESS', result)
  } catch (error) {
    context.commit('DATA_ERROR', error)
  }

}
```

What do you think?